amazon-kinesis-client-multilang/src/test/java/software/amazon/kinesis/multilang/MultiLangDaemonConfigTest.java

```diff
@@ -35,62 +31,168 @@
 import software.amazon.awssdk.auth.credentials.AwsCredentials;
 import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
 import software.amazon.kinesis.multilang.config.KinesisClientLibConfigurator;
-import software.amazon.kinesis.multilang.config.MultiLangDaemonConfiguration;
 
 @RunWith(MockitoJUnitRunner.class)
 public class MultiLangDaemonConfigTest {
-    private static String FILENAME = "some.properties";
+    private static final String TEST_FILENAME = "some.properties";
+    private static final String TEST_EXE = "TestExe.exe";
+    private static final String TEST_APPLICATION_NAME = "TestApp";
+    private static final String TEST_STREAM_NAME = "fakeStream";
+    private static final String TEST_STREAM_NAME_IN_ARN = "FAKE_STREAM_NAME";
+    private static final String TEST_REGION = "us-east-1";
+    private static final String TEST_STREAM_ARN = "arn:aws:kinesis:us-east-2:012345678987:stream/" + TEST_STREAM_NAME_IN_ARN;
+
+    @Mock
+    ClassLoader classLoader;
 
     @Mock
     private AwsCredentialsProvider credentialsProvider;
     @Mock
     private AwsCredentials creds;
-    @Mock
+
     private KinesisClientLibConfigurator configurator;
+    private MultiLangDaemonConfig deamonConfig;
+
+    /**
+     * Instantiate a MultiLangDaemonConfig object based on the properties passed in.
+     * @param streamName
+     * @param streamArn
+     * @param regionName
+     * @throws IOException
+     */
+    public void setup(String streamName, String streamArn, String regionName) throws IOException {
+
+        String properties = String.format("executableName = %s\n"
```

Comment:
[`StringBuilder`](https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html) should be objectively faster, and may produce code that is subjectively easier to read.

StringBuilder would also provide a single mechanism versus `String.format(...)` _plus_ `concat` (lines ~75, 78).